Kickstarter-iOS/Views/LoadingButton.swift

```diff
@@ -0,0 +1,115 @@
+import Library
+import Prelude
+import UIKit
+
+final class LoadingButton: UIButton {
+  // MARK: - Properties
+
+  private lazy var activityIndicator: UIActivityIndicatorView = {
+    UIActivityIndicatorView(style: .white)
+      |> \.hidesWhenStopped .~ true
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  public var isLoading: Bool = false {
+    didSet {
+      guard self.isLoading != oldValue else { return }
+
+      if self.isLoading {
+        self.startLoading()
+      } else {
+        self.stopLoading()
+      }
+
+      _ = self
+        |> \.isUserInteractionEnabled .~ !self.isLoading
+    }
+  }
+
+  private var originalTitles: [UInt: String] = [:]
+
+  // MARK: - Lifecycle
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+
+    self.addSubview(self.activityIndicator)
+    self.activityIndicator.centerXAnchor.constraint(equalTo: self.centerXAnchor).isActive = true
+    self.activityIndicator.centerYAnchor.constraint(equalTo: self.centerYAnchor).isActive = true
+  }
+
+  required init?(coder: NSCoder) {
+    super.init(coder: coder)
+
+    self.addSubview(self.activityIndicator)
+    self.activityIndicator.centerXAnchor.constraint(equalTo: self.centerXAnchor).isActive = true
+    self.activityIndicator.centerYAnchor.constraint(equalTo: self.centerYAnchor).isActive = true
+  }
+
+  override func setTitle(_ title: String?, for state: UIControl.State) {
+    // Do not allow changing the title while the activity indicator is animating
+    guard !self.activityIndicator.isAnimating else { return }
```

Comment:
Setting a title while the activity indicator is loading would display both the indicator and the title so I think this is the right approach here..we can potentially add logic to stop animating if we're setting title but I think due to our RAC bindings it could cause some flickering